### PR TITLE
Append ampersand if entryPoint or logoutUrl has a query string

### PR DESCRIPTION
**Pull request by amoro99**

### CODE REVIEW

File: lib/saml.js [hunk 1]
```diff
@@ -161,11 +161,16 @@ SAML.prototype.requestToUrl = function (request, operation, callback) {
...
+        var hasQuery = function(url) {
+            return !!(new URL(url).search);
+        }
+
+        var target = self.options.entryPoint + (hasQuery(self.options.entryPoint) ? '&' : '?');
 
         if (operation === 'logout') {
             if (self.options.logoutUrl) {
-                target = self.options.logoutUrl + '?';
+                target = self.options.logoutUrl + (hasQuery(self.options.logoutUrl) ? '&' : '?');
```
Comment (derek-oneill) [inline lib/saml.js]: A more robust way to do this is to use the URL builder itself. https://nodejs.org/docs/latest-v10.x/api/url.html#url_class_urlsearchparams

**Review by derek-oneill — CHANGES_REQUESTED** [review]
See line comment

File: test/module-spec.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+            expect(parsed.query.Signature).to.not.be.empty;
+        })
+
+    });
+    it(`should behave normally if the idp url has no query string`, () => {
+        const saml = new SAML({
+            protocol: 'https://',
+            path: 'someidp.com/saml/2.0/auth',
+            callbackUrl: 'https://generalsoftware.com/auth',
+            entryPoint: 'https://someidp.com/saml/2.0/auth?id=foo',
```
Comment (derek-oneill) [inline test/module-spec.js]: This still includes the query param

Comment (amoro99) [reply test/module-spec.js]: Ok Made change

**Review by derek-oneill — APPROVED** [review]

**Pull request merged**